definitions/features/downstream.rb

```diff
@@ -55,24 +55,32 @@ def rhsm_refresh
   def rh_repos(sat_version)
     sat_version = version(sat_version)
     rh_version_major = execute!('facter operatingsystemmajrelease')
-    sat_version_full = "#{sat_version.major}.#{sat_version.minor}"
+    rh_repos = main_rh_repos(rh_version_major)
 
+    sat_version_full = "#{sat_version.major}.#{sat_version.minor}"
     sat_repo_id = "rhel-#{rh_version_major}-server-satellite-#{sat_version_full}-rpms"
     sat_tools_repo_id = "rhel-#{rh_version_major}-server-satellite-tools-#{sat_version_full}-rpms"
-    # Override to use Beta repositories for 6.4 until GA
+
     if sat_version.to_s == '6.4'
+      # Override to use Beta repositories for 6.4 until GA
       sat_repo_id = "rhel-server-#{rh_version_major}-satellite-6-beta-rpms"
+
+      rh_repos << 'rhel-7-server-ansible-2-rpms'
     end
+    rh_repos.concat([sat_tools_repo_id, sat_repo_id])
 
-    rh_repos = ["rhel-#{rh_version_major}-server-rpms",
-                "rhel-server-rhscl-#{rh_version_major}-rpms",
-                "rhel-#{rh_version_major}-server-satellite-maintenance-6-rpms",
-                sat_tools_repo_id,
-                sat_repo_id]
-    rh_repos << 'rhel-7-server-ansible-2-rpms' if sat_version.to_s == '6.4'
+    if current_minor_version == '6.3' && (find_package('puppet') =~ /puppet-4/)
```

Comment:
Moving the logic to the puppet feature may make it more readable and easier to change
```ruby
if current_minor_version == '6.3' && feature(:puppet) && feature(:puppet).version.major == 4
```
also on none of my instances (sat 6.4, katello 3.6) the package `puppet` is not installed. 
`puppet --version` works everywhere. Might be more flexible to use something like `version(execute!('puppet --version'))`